source/Account.cpp

```diff
@@ -224,6 +224,19 @@ string Account::Step(int64_t assets, int64_t salaries, int64_t maintenance)
 	// makes you at least as credit-worthy as someone who pays debts on time.)
 	creditScore = max(200, min(800, creditScore + (missedPayment ? -5 : 1)));
 	
+	auto creditString = [](int64_t payment) -> string
+	{
+		return payment == 1 ? "1 credit" : Format::Credits(payment) + " credits";
+	};
+	
+	if(maintenancePaid < 0)
+	{
+		if(missedPayment)
+			out << " ";
+		out << "You received " << creditString(-maintenancePaid) << " in maintenance.";
+		maintenancePaid = 0;
+	}
+	
 	// If you didn't make any payments, no need to continue further.
 	if(!(salariesPaid + maintenancePaid + mortgagesPaid + finesPaid))
```

Comment:
MaintenancePaid could now cancel out the others and doesn't need to be handled when negative;

```suggestion
	if(!salariesPaid && (maintenancePaid <= 0) && !mortgagesPaid &&!finesPaid)
```

Reply:
That's why I set maintenancePaid to 0 after saying how much you received so that we don't need to handle the negative case for payments.